queries/document.py

```diff
@@ -141,26 +143,37 @@ def construct_attach_document_to_unsigned_version(doc_uri, prev_ver_doc_uri, gra
         $doc sign:ongetekendStuk $prev_doc .
         $doc dct:title ?prev_title .
         ?case dossier:Dossier.bestaatUit $doc .
-        $doc besluitvorming:vertrouwelijkheidsniveau $access_level .
+        $doc besluitvorming:vertrouwelijkheidsniveau ?access_level .
     }
 }
 WHERE {
     GRAPH $graph {
         $doc a dossier:Stuk .
         $prev_doc a dossier:Stuk ;
-            dct:title ?prev_title .
+            dct:title ?prev_title ;
+            besluitvorming:vertrouwelijkheidsniveau ?prev_access_level .
         OPTIONAL {
             $doc dct:title ?old_title .
         }
         OPTIONAL {
             ?case a dossier:Dossier ;
                 dossier:Dossier.bestaatUit $prev_doc .
         }
+        BIND(
+          COALESCE(
+            IF(?prev_access_level = $access_level_public, $access_level_cabinet, 1/0),
+            IF(?prev_access_level = $access_level_government, $access_level_cabinet, 1/0),
+            ?prev_access_level
+          ) AS ?access_level
+        )
```

Comment:
Would `BIND(IF(?prev_access_level IN ($access_level_public, $access_level_government), $access_level_cabinet, ?prev_access_level) AS ?access_level)` be a more readable expression than this one featuring the divide by zero? :-)

Reply:
I think that would be equivalent... Would need to test though

Reply:
All fine by me, for context I used this here after Tom linked me this

https://semaku.com/post/using-coalesce-and-if-in-sparql-for-nested/

Specifically the usage of `COALESCE` being faster, and the `1/0` being used so that the falsy return of the `IF`s is null and we keep cascading down the  conditions.

But the `?thing IN (list)` could be just as fast, not sure.

Reply:
Was purely a readability-thing. Sleeping on it for a night got my eyes used to the `1/0` ;-) Will merge as-is.

Reply:
So ... we ran this on the server and Virtuoso surprised us with `Virtuoso 22012 Error SR084: Division by 0.` :upside_down_face: